### PR TITLE
Adding Emails to New Cohort Members to Finish Signup

### PR DESCRIPTION
**Pull request by brvarner**
- Fixes Cohorts => Cohort_Members forms as noticed by Reza
- Adds Cohort Member Mailer to app/mailers
  - Mailer contains two actions to correspond to welcoming mentors or mentees
- Updates Application Mailer to use 'onboarding@resend.dev' for testing, will update our email address in the future once we have a verified domain
- Adds Resend gem to Gemfile
- Adds Resend API Key to Rails Credentials file. We may update this credential after creating another account just for this project
- Adds mailer initializer to the project, it provides the mailer with our API Key
- Added mailer action to Cohort Member model in Arpan's function to send appropriate emails to new cohort members based on role. May add more logic later to differentiate between existing users and new users

### CODE REVIEW

File: app/views/cohort_member_mailer/mentee_welcome_mail.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<h1>Welcome to MentE!</h1>
+
+<p>You're registered as a mentee for Discovery Partners Institute's <%= @cohort.program.name %> <%= @cohort.cohort_name %>!</p>
+
+<p>Please follow this link to register your account: insert link.</p>
+
+<p>Yours Truly,</p>
+<p>The MentE Team</p>
```
Comment (MayaS1111) [inline app/views/cohort_member_mailer/mentee_welcome_mail.html.erb:8]: Cute!

**Review by MayaS1111 — APPROVED** [review]
Create work Brandon

File: app/views/cohort_member_mailer/mentee_welcome_mail.html.erb
Comment (APatel-AI) [inline app/views/cohort_member_mailer/mentee_welcome_mail.html.erb:1]: Details 👌

**Review by APatel-AI — APPROVED** [review]
Great Job Brandon!

**Pull request merged**